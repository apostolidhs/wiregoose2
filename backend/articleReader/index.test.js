const path = require('path');
const {fromFile} = require('./index');

const articles = {
  madata: '../../../fixtures/articles/madata.htm',
  newsit: '../../../fixtures/articles/newsit.htm',
  altsantiri: '../../../fixtures/articles/altsantiri.htm'
};

Object.entries(articles).forEach(([name, pathname]) => {
  it(`reads ${name}`, async () => {
    const content = await fromFile(path.resolve(__filename, pathname));
    console.log(content);
    expect(content).toMatchSnapshot();
  });
});
