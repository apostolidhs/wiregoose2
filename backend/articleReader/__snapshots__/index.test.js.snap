// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reads altsantiri 1`] = `
Array [
  Object {
    "content": "Ο Κωστής Σαββιδάκης υποδύεται τον παππά στη σειρά Άγριες Μέλισσες και γνωρίζει μεγάλη απήχηση.",
    "type": "p",
  },
  Object {
    "content": "Ο ηθοποιός την Πέμπτη φιλοξενήθηκε στην εκπομπή Το πρωινό μιλώντας για τη σειρά αλλά και για τη ζωή του.",
    "type": "p",
  },
  Object {
    "src": "https://www.altsantiri.gr/wp-content/uploads/2020/01/vlcsnap-00027.jpg",
    "type": "img",
  },
  Object {
    "content": "Με έχει βαφτίσει παπάς, η γυναίκα του η παπαδιά και ο γιος του, ήμουν από μικρός ως παπαδάκι, στο κατηχητικό. Είμαι θρήσκος, θυμιατίζω σχεδόν κάθε μέρα στο σπίτι, ρίχνω αγιασμό. Δεν ντρέπομαι να το πω, γιατί δυστυχώς, πολλοί έχουμε φτάσει στο σημείο να ντρεπόμαστε να πούμε ότι είμαστε Χριστιανοί Ορθόδοξοι. Σέβομαι την κάθε θρησκεία και θέλω να σέβονται και τη δικιά μου θρησκεία. Νομίζω ότι προσπαθούν να μας απομακρύνουν από τα ήθη και τα έθιμά μας, τη θρησκεία μας, ειδικά εμάς τους Έλληνες.",
    "type": "p",
  },
  Object {
    "src": "https://www.altsantiri.gr/wp-content/uploads/2020/01/vlcsnap-00029.jpg",
    "type": "img",
  },
  Object {
    "content": "Δεν μπορώ να επιτρέψω σε οποιονδήποτε να μπει σε ένα μοναστήρι να κάψει εικόνες, να σπάσει σταυρούς. Θα μπω να τον κουτουλήσω. Όπως ένας μουσουλμάνος θα μου κόψει το κεφάλι αν πειράξω τη θρησκεία του και το σέβομαι, έτσι πρέπει να λειτουργούμε κι εμείς. Όχι να τον σφάξουμε, αλλά λες λίγο ένα όριο. Είναι η πίστη μου, η θρησκεία μου.",
    "type": "p",
  },
  Object {
    "content": "ΣΧΟΛΙΑ",
    "type": "h",
  },
  Object {
    "content": "Tο altsantiri.gr δημοσιεύει άμεσα κάθε σχόλιο. Ωστόσο δεν υιοθετούμε τις απόψεις αυτές καθώς εκφράζουν αποκλειστικά τον εκάστοτε σχολιαστή. Σχόλια με ύβρεις διαγράφονται χωρίς προειδοποίηση. Χρήστες που δεν τηρούν τους όρους χρήσης αποκλείονται.",
    "type": "p",
  },
]
`;

exports[`reads madata 1`] = `
Array [
  Object {
    "content": "Ένα τεστ αίματος που αποκαλύπτει αν κάποιος κοιμάται αρκετά, αναπτύσσουν Ιταλοί επιστήμονες. Το τεστ θα βοηθήσει, μεταξύ άλλων, τους γονείς να γνωρίζουν κατά πόσο το παιδιά τους πάσχουν από χρόνια στέρηση ύπνου.",
    "type": "p",
  },
  Object {
    "content": "Οι ερευνητές, με επικεφαλής τον δρ Φάμπιο Λάουρα του Ινστιτούτου Επιστημών της Διατροφής του Εθνικού Συμβουλίου Ερευνών της Ιταλίας, που έκαναν τη σχετική δημοσίευση στο περιοδικό πειραματικής φυσιολογίας Experimental Physiology, βασίζουν το τεστ τους στην ανακάλυψη ότι μικροσκοπικά τμήματα του RNA (του ξαδέρφου μορίου του DNA, το οποίο μεταφράζει το γενετικό κώδικα σε πρωτεΐνες) μεταβάλλονται στο αίμα των ανθρώπων που δεν κοιμούνται αρκετά.",
    "type": "p",
  },
  Object {
    "content": "Η μελέτη μας δείχνει για πρώτη φορά ότι τα επίπεδα συγκεκριμένων microRNA που κυκλοφορούν στο αίμα, είναι διαφορετικά σε όσους κοιμούνται λίγο, σε σχέση με όσους κοιμούνται φυσιολογικά. Αυτό θα επιτρέψει στους γιατρούς να προσδιορίζουν εύκολα κατά πόσο τα παιδιά κοιμούνται αρκετά, χρησιμοποιώντας ένα απλό τεστ αίματος, που θα παρέχει ενδείξεις και για άλλες όψεις της υγείας, δήλωσε ο δρ Λάουρα.",
    "type": "p",
  },
  Object {
    "content": "Προηγούμενες έρευνες έχουν δείξει ότι ο ύπνος είναι ουσιώδης για την υγεία και τις σχολικές επιδόσεις των παιδιών, ενώ αργότερα στη ζωή η έλλειψη του σχετίζεται με διάφορα προβλήματα υγείας.",
    "type": "p",
  },
  Object {
    "content": "Οι ερευνητές μελέτησαν δείγματα αίματος από 111 υγιή παιδιά και εφήβους από οκτώ ευρωπαϊκές χώρες (μεταξύ των οποίων η Κύπρος), που χωρίστηκαν σε δύο ομάδες, του επαρκούς και τους ανεπαρκούς ύπνου. Ως ανεπαρκής για τα παιδιά θεωρήθηκε ο ύπνος κάτω των εννέα ωρών και για τους εφήβους κάτω των οκτώ ωρών.",
    "type": "p",
  },
  Object {
    "content": "Διαπιστώθηκε ότι -άσχετα με τη χώρα προέλευσης- είναι δυνατό, αναλύοντας τα microRNA στο αίμα, να διακρίνει κανείς με αρκετή αξιοπιστία αν κάποιος κοιμάται αρκετά ή όχι. Θα χρειαστεί πάντως κάποιος χρόνος, εωσότου υπάρξει διαθέσιμο ένα κανονικό κλινικό τεστ.",
    "type": "p",
  },
]
`;

exports[`reads newsit 1`] = `
Array [
  Object {
    "content": "Δημοσιεύθηκε: 09.01.2020 | 18:29",
    "type": "p",
  },
  Object {
    "content": "Drive -",
    "type": "p",
  },
  Object {
    "src": "data:image/svg+xml,%3Csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%200%200'%3E%3C/svg%3E",
    "type": "img",
  },
  Object {
    "src": "data:image/svg+xml,%3Csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%200%200'%3E%3C/svg%3E",
    "type": "img",
  },
  Object {
    "content": "Διαθέσιμα από το πρώτο τρίμηνο της χρονιάς τα ήπια υβριδικά της μάρκας που είναι το πρώτο βήμα για τον εξηλεκτρισμό της.",
    "type": "h",
  },
  Object {
    "content": "Τα FΙΑΤ 500 και FΙΑΤ Panda είναι εμβληματικά μοντέλα στην κατηγορία των αυτοκινήτων πόλης στην Ευρωπαϊκή, αλλά και την Ελληνική αγορά αυτοκινήτου. Αυτά τα δύο μοντέλα επέλεξε ο Όμιλος Fiat Chrysler Automobiles για να εφαρμόσει μια νέα υβριδική τεχνολογία. Τα νέα 500 Hybrid και Panda Hybrid, τα οποία θα ξεκινήσουν να είναι διαθέσιμα το 1ο τρίμηνο του έτους, θα φέρουν επανάσταση για ακόμα μία φορά στις αστικές μετακινήσεις, κάνοντας προσιτή την υβριδική τεχνολογία σε όλους.",
    "type": "p",
  },
  Object {
    "content": "Διαβάστε περισσότερα στο drive.gr",
    "type": "p",
  },
]
`;
